test/components/converters/test_pypdf_to_document.py

```diff
@@ -84,32 +65,10 @@ def test_to_dict(self, pypdf_component):
             },
         }
 
-    def test_to_dict_custom_converter(self):
-        pypdf_component = PyPDFToDocument(converter=CustomConverter(), store_full_path=False)
-        data = pypdf_component.to_dict()
-        assert data == {
-            "type": "haystack.components.converters.pypdf.PyPDFToDocument",
-            "init_parameters": {
-                "converter": {
-                    "data": {"key": "value", "more": False},
-                    "type": "converters.test_pypdf_to_document.CustomConverter",
-                },
-                "extraction_mode": "plain",
-                "plain_mode_orientations": (0, 90, 180, 270),
-                "plain_mode_space_width": 200.0,
-                "layout_mode_space_vertically": True,
-                "layout_mode_scale_weight": 1.25,
-                "layout_mode_strip_rotated": True,
-                "layout_mode_font_height_weight": 1.0,
-                "store_full_path": False,
-            },
-        }
-
     def test_from_dict(self):
```

Comment:
Can you quickly test what happens if deserialized `PyPDFToDocument` with a Converter can be read with this new code? It seems so, but just to be sure

Reply:
You mean something like this? (passing)

```python
def test_from_dict_skip_custom_converter(self):
        data = {
            "type": "haystack.components.converters.pypdf.PyPDFToDocument",
            "init_parameters": {
                "converter": {
                    "data": {"key": "value", "more": False},
                    "type": "converters.test_pypdf_to_document.CustomConverter",
                }
            },
        }

        with pytest.raises(TypeError) as exc:
            PyPDFToDocument.from_dict(data)

        assert "PyPDFToDocument.__init__() got an unexpected keyword argument 'converter'" in str(exc.value)
```

Having a `converter` in `init_parameters` will make code to throw currently. Is this ok or do we want to e.g. ignore it?

Reply:
as discussed offine for this particular use case it's OK to fail